include/openthread/ip6.h

```diff
@@ -473,11 +474,43 @@ bool otIp6IsReceiveFilterEnabled(otInstance *aInstance);
  * @param[in]  aEnabled  TRUE if Thread control traffic is filtered out, FALSE otherwise.
  *
  * @sa otIp6SetReceiveCallback
- * @sa otIsReceiveIp6FilterEnabled
+ * @sa otIp6SetReceiveFullFilterEnabled
+ * @sa otIp6IsReceiveFilterEnabled
  *
  */
 void otIp6SetReceiveFilterEnabled(otInstance *aInstance, bool aEnabled);
 
+/**
+ * This function sets whether or not any Thread initiated traffic is filtered out when delivering IPv6 datagrams
+ * via the callback specified in otIp6SetReceiveCallback(). This filters out for example CoAP communication
+ * if OpenThread CoAP API is used.
+ *
+ * @param[in]  aInstance A pointer to an OpenThread instance.
+ * @param[in]  aEnabled  TRUE if Thread initiated traffic is filtered out, FALSE otherwise.
+ *
+ * @sa otIp6SetReceiveCallback
+ * @sa otIp6IsReceiveFullFilterEnabled
+ * @sa otIp6SetReceiveFilterEnabled
+ *
+ */
+void otIp6SetReceiveFullFilterEnabled(otInstance *aInstance, bool aEnabled);
```

Comment:
Thanks, I was considering this, however decided not to do that. This would make API incompatible with previous one, and every project using it would need to be modified a bit. If you think that's ok, I can change it.

Reply:
While I agree we should generally try to keep the existing API stable, in this particular case, I am fine to change the API. I don't think the `otIp6GetReceiveFilterMode()` and `otIp6SetReceiveFilterMode()` APIs are used in many places and it should be easy to resolve any conflicts that exist.

Reply:
I agree. In general we want to keep exiting API but in this  case, it may be fine to allow the API change to keep the number of APIs smaller and easier to use (I also expect that this API is not being used in many places/projects).